File: UserStore.cpp
#include "UserStore.h"
#include <algorithm>

void UserStore::addUser(const User& user) {
    users.push_back(user);
}

User* UserStore::findByName(const std::string& name) {
    for (auto& u : users) {
        if (u.getName() == name) {
            return &u;
        }
    }
    return nullptr;
}

void UserStore::removeByName(const std::string& name) {
    users.erase(
        std::remove_if(users.begin(), users.end(),
            [&name](const User& u) { return u.getName() == name; }),
        users.end()
    );
}

void UserStore::printAll() const {
    for (const auto& u : users) {
        u.printInfo();
    }
}

int UserStore::count() const {
    return static_cast<int>(users.size());
}

File: user_store.py
from user import User


class UserStore:
    def __init__(self):
        self.users: list[User] = []

    def add_user(self, user: User):
        self.users.append(user)

    def find_by_name(self, name: str):
        for u in self.users:
            if u.get_name() == name:
                return u
        return None

    def remove_by_name(self, name: str):
        self.users = [u for u in self.users if u.get_name() != name]

    def print_all(self):
        for u in self.users
            u.print_info()

    def count(self) -> int:
        return len(self.users)
